### PR TITLE
Added test showing that the decimal decode error.

### PR DESCRIPTION
**Pull request by gdey**
I am getting some very very small numbers instead of the expected
numbers. The issue turned out to be the with the inital byte. When
there was a negative number that byte and the number was small enough
to fit in one byte, the initial byte size would be 1. We were then using
a 4 byte mask to mask out 1 byte value leaving the mask as part of the
value, causing the error. This was only an issue when the number was
negative as the mask was the 2's complement. For positive number the
mask is a zero, so it was not contributing to the final value.

I've added test cases that show the errors (the negative values) as well as test cases that were working.

### CODE REVIEW

File: replication/row_event_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package replication
+
+import (
+	"testing"
+)
+
+func TestDecodeDecimal(t *testing.T) {
```
Comment (siddontang) [inline replication/row_event_test.go]: We should use gocheck instead of origin testing.T like other test cases in go-mysql. :-)

**Comment by siddontang** [comment]
Thanks @gdey 

How to reproduce the bug in mysql? what's the table schema, and the insert decimal data?

File: replication/row_event_test.go (hunk 1 above)
Comment (gdey) [inline replication/row_event_test.go]: Okay, I can do that. I'll rewrite the tests to use gocheck. Like the others.

**Comment by gdey** [comment]
I'll have to build a simpler schema and put in the offending values. the column type was decimal(13,2) and the values are the values in the test, that were causing the issue.

**Comment by siddontang** [comment]
Hi @gdey 

In our replication test, we use `de DECIMAL(10,2),` , can you help me add another `decimal (13, 2)` in the test?

**Comment by gdey** [comment]
@siddontang, Yes, also, I noticed a bunch of more error regarding the decode decimal. I'm working on a new pull request. I'm hoping to have that to you soon. Just want to make a comment not to merge this pull request. yet.

**Comment by siddontang** [comment]
Thanks @gdey 

Hope your new PR soon :-)

**Comment by gdey** [comment]
@siddontang I looked at replication_test.go, I'll see look at adding the test in there. But the decimal(10,2) and decimal(13,2) values can be the same. After all we are just increasing the precision of the decimal value. So, I think we can just add a new field and use the same values. I'll add a new commit soon.

**Comment by gdey** [comment]
As far as the t directory; I can remove it. In there I have everything I used to generate the test data. The only thing that is really needed is the gensql.go file or the ddsql.sql file. So I'm happy to remove it and add comments to the main test file on how to generate the test data. I think that makes the most sense. that way if we find more value there is a record of how to add more values to the test cases.

**Comment by siddontang** [comment]
LGTM

Can we merge it now? @gdey

**Comment by gdey** [comment]
Yes

**Comment by siddontang** [comment]
Cool

**Pull request merged**